docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,90 @@
+= Git Submodule pattern
+
+== How this workflow pattern works:
+
+1. Create a "downstream" repo.
+2. Track your upstream git repo with a git submodule reference.
+3. Onboard to {ProductName}, use link:https://github.com/renovatebot/renovate[Renovate] to keep your Git submodule references up to date.
+4. In your downstream repo, apply patches in your Containerfile if you need to carry downstream patches.
+
+== Implementing this solution pattern
+=== Create a downstream repo
+
+A **downstream repo** is a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+The downstream repo will:
+
+* be used to hold the git submodule reference to the upstream repository.
+* contain a Containerfile that references and pulls in the git submodule(s) content.
+* become a Component of our {ProductName} application.
+
+=== Create the git submodule for the upstream repo to track it
+
+An **upstream repo** is a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+After creating the downstream repo, add a git submodule for the upstream repo:
+
+[source, bash]
+----
+git submodule add <upstream-url>
+----
+
+Check and configure the git submodule config:
+[source, gitmodules]
+----
+# .gitmodules
+[submodule "<project-name>"]
+  path = <path-in-downstream-repo>
+  url = https://github.com/<namespace>/<project>.git
+  branch = main # Ensure the branch is correct
+----
+See the link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on configuring git submodules.
+
+=== Onboard the project onto {ProductName}
+
+After creating the repository, onboard to {ProductName} and configure your application and its components.
+
+**Note:**
+The xref:/getting-started/index.adoc[Getting started docs] are currently geared towards configuring YAML files.
+____
+At the time of publication, to create applications in {ProductName}, you need to manually configure them by editing YAML files
+____
+
+This repo was configured using the {ProductName} UI.
+There does not seem to be any detailed steps on how to do this yet via the UI.
+
+Here are some **brief** steps on onboarding to {ProductName} using the UI:
+
+1. Create an application in your workspace.
+2. Configure your application component.
+3. If you're using Github, ensure that you have the link:https://github.com/apps/red-hat-konflux[Red Hat {ProductName} bot] enabled.
+4. Check your {ProductName} CI results. Ensure that your component was able to be pulled from, built and tested.
+
+For more information on configuring creating applications and components, please read xref:/how-tos/creating.adoc[this documentation].
```

Comment:
It is not currently possible to onboard with the UI due to the fact that we do not have a means to create the image repositories. This is a known gap with the upstream project and our own implementation. While it is unfortunate that we are not documenting this flow, I think we can leave it out for now.

Reply:
removed from doc